zebra-network/src/address_book.rs

```diff
@@ -3,8 +3,9 @@
 
```

Comment:
The title of this PR needs a tweak, because the `AddressBook` only stores the addresses of outbound peer connections.

So this PR only prevents rapidly repeated outbound reconnections and handshakes to the same IP. (Which is the attack I'm trying to address, and what I should have written in the ticket.)

It's ok to leave any duplicate outbound/inbound IPs for the peer set to drop. That would be a different attack, which seems much less plausible to me. But if you can think of a high-impact attack across inbound and outbound connections, please open a ticket!